COMETwebapp/Components/SubscriptionDashboard/SubscribedTable.razor

```diff
@@ -92,20 +93,47 @@ Copyright (c) 2022 RHEA System S.A.
     /// </summary>
     public ObservableCollection<ParameterSubscriptionViewModel> ViewModelSubscribedParameters = new ObservableCollection<ParameterSubscriptionViewModel>();
 
+    /// <summary>
+    /// Listeners for the components to update it with ISession
+    /// </summary>
+    private Dictionary<string, IDisposable> Listeners = new Dictionary<string, IDisposable>();
+
     /// <summary>
     /// Initialize component at first render and after session update
     /// </summary>
     protected override void OnInitialized()
     {
-        this.SubscribedParameters = iterationService.GetParameterSubscriptions(ISessionAnchor.OpenIteration, ISessionAnchor.CurrentDomainOfExpertise);
+        this.ViewModelSubscribedParameters.Clear();
+        this.SubscribedParameters = iterationService.GetParameterSubscriptions(ISessionAnchor.GetIteration(), ISessionAnchor.CurrentDomainOfExpertise);
         this.GetViewModelSubscribedParameters();
+    }
 
-        CDPMessageBus.Current.Listen<SessionEvent>().Where(x => x.Status == SessionStatus.EndUpdate).Subscribe(x => {
-            this.SubscribedParameters = iterationService.GetParameterSubscriptions(ISessionAnchor.OpenIteration, ISessionAnchor.CurrentDomainOfExpertise);
-            this.ViewModelSubscribedParameters.Clear();
-            this.GetViewModelSubscribedParameters();
-            this.StateHasChanged();
-        });
+    protected override void OnAfterRender(bool firstRender)
+    {
+        if (this.Listeners.TryGetValue("UpToDate", out var listener))
```

Comment:
I think you need to reverse the logic here `if (!this.Listeners.TryGetValue("UpToDate", out var listener))`